File: apps/ios/NOVA/Models/Alert.swift
import Foundation

/// A project changing state while nobody was looking.
struct Alert: Decodable, Identifiable, Equatable, Sendable {
    enum Kind: String, Decodable, Sendable {
        case projectDown = "project_down"
        case projectRecovered = "project_recovered"
    }

    let id: UUID
    let kind: Kind
    let project: String
    let message: String
    let createdAt: Date
    let acknowledgedAt: Date?

    var isAcknowledged: Bool { acknowledgedAt != nil }
    var isDown: Bool { kind == .projectDown }
}

struct AlertPage: Decodable, Equatable, Sendable {
    let items: [Alert]
    /// Across every alert, not only this page.
    let unacknowledged: Int
}

/// The dashboard's summary of the watcher.
struct AlertsStatus: Decodable, Equatable, Sendable {
    let watching: Bool
    let intervalSeconds: Int
    let unacknowledged: Int
    let latest: Alert?
}

struct AcknowledgedCount: Decodable, Sendable {
    let acknowledged: Int
}

File: services/api/src/nova/schemas/alert.py
"""Alerts as the app reads them."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AlertRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    kind: str = Field(description="project_down or project_recovered.")
    project: str
    message: str
    created_at: datetime
    acknowledged_at: datetime | None = None


class AlertPage(BaseModel):
    items: list[AlertRead]
    unacknowledged: int = Field(description="Across every alert, not only this page.")


class AcknowledgedCount(BaseModel):
    acknowledged: int
